LibertyDSNP/example-client-frequency
# Bugs/signature verification

## Problem
Signature verification for calling `createSponsoredAccountWithDelegation` was failing on chain. 

## Solution
In the `scale encode` function, we were putting the arguments in the wrong order and the "account" (really the authorized_msa_id) needed to be 64 bits instead of 32 bits.

Also renamed some things to make what they are/do more clear.

## Steps to verify:
1. Make sure Local relay and parachains are running
2. Make sure you have Polkadot JS extension installed and at least one account
3. `yarn start`
4. Wait for the chain to be connected and the service MSA ID to appear - there should be 3 green boxes at the bottom of the page. You may have to reload a couple of times and watch the Polkadot dashboard for the "msa.create" event (it's a demo, it's not robust), and you may see some "priority too low" alerts. 
5. Login as one of the accounts shown in the list.
6. Click the "Register MSA" button at the bottom left
7. Wait a bit for the transaction to be in a block. The extrinsic should not have failed.
8. Eventually the MSA ID should appear in the row that you're logged in as.  If not, reload the page.
